sklearn/neighbors/binary_tree.pxi

```diff
@@ -2121,21 +2153,38 @@ cdef class BinaryTree:
                 global_log_bound_spread =\
                     logsubexp(global_log_bound_spread,
                               node_log_bound_spreads[i_node])
-                for i in range(node_info.idx_start, node_info.idx_end):
-                    dist_pt = self.dist(pt, data + n_features * idx_array[i],
-                                        n_features)
-                    log_density = compute_log_kernel(dist_pt, h, kernel)
-                    global_log_min_bound = logaddexp(global_log_min_bound,
-                                                     log_density)
+                if with_sample_weight:
```

Comment:
Does all this branching provide much benefit?

Reply:
It looks more maintainable to me, which is my concern. Near-duplicate code is hard to see differences in

Reply:
I see your point, I'll do some benchmarking to see if this branching is a actually improving anything.

Reply:
Removing the branching slows down the fit method (both with and without sample weights unfortunately) by about 10-15%, for 10^5 sample with 20 columns on my local machine.

Reply:
I committed the simplified code anyway. Happy to revert the commit if we want to avoid the consequent slowdown.